Strong Number Using Indirect recursion

ErzaTitania-2001 opened the pull request:
<!--
Thank you for sending the PR! We appreciate you spending the time to work on these changes.

Help us understand your motivation by explaining why you decided to make this change.
You can learn more about contributing to NeoAlgo here: https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md

Happy Contributing!

-->

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes
### Description

(Write your answer here.)

### Checklist

- [ yes] I've read the contribution guidelines.
- [yes ] I've checked the issue list before deciding what to submit.
- [ yes] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Math Programs (WoC) #1527

ErzaTitania-2001 committed:
Strong Number Using Indirect Recursion

Math Programs (WoC) #1527

ErzaTitania-2001 committed:
Updated Indirect Recursion for Strong Number

Math Programs (WoC) #1527

ErzaTitania-2001 committed:
Update README.md

ErzaTitania-2001 commented:
@iamrajiv  check my PR for any discrepancies and inform me if any! 
Issue in question Math Programs (WoC) #1527

HarshCasper commented on Java/math/StrongRecursion.java:
```diff
@@ -0,0 +1,47 @@
+//checking for a strong number using indirect recursion
```
Add a description of the Algorithm/Code that you are looking to implement

HarshCasper commented on Java/math/StrongRecursion.java:
```diff
@@ -0,0 +1,47 @@
+import java.util.*;
+class StrongRecursion
+{
+    public static int sofd(int n) //recursive function for sum of factorial of digits
```
Keep the Comments above the Line of Code not beside it

HarshCasper replied:
Use meaningful function and variable names

HarshCasper commented on Java/math/StrongRecursion.java:
```diff
@@ -0,0 +1,47 @@
+            return n;
+        else
+            return (factorial(n%10)+sofd(n/10));
+
```
Remove redundant lines of code

HarshCasper commented on Java/math/StrongRecursion.java:
```diff
@@ -0,0 +1,47 @@
+
+    public static void main()
+    {
+        Scanner sc=new Scanner(System.in);
```
You can try making the User Input more friendly

HarshCasper commented on Java/math/StrongRecursion.java:
```diff
@@ -0,0 +1,47 @@
+ *  
+ * 
+ */
+ 
```
Add a newline at the end of the file

ErzaTitania-2001 committed:
Updated file

Added description of algorithm/code
Changed function name
Modified comment lines
Made user input more friendly
Added new line at the EOF

ErzaTitania-2001 reviewed (commented):
Changes done @HarshCasper . Go over it.

HarshCasper commented on Java/math/StrongRecursion.java:
```diff
@@ -0,0 +1,56 @@
+ * 
+ */
+
+
```
Remove extra lines of code

HarshCasper commented on Java/math/StrongRecursion.java:
```diff
@@ -0,0 +1,56 @@
+    }
+}
+
+/* Sample Input And Output ; 
```
`:` not `;`

HarshCasper commented on Java/math/StrongRecursion.java:
```diff
@@ -0,0 +1,56 @@
+    public static int sumOfFactorialOfDigit(int n) 
+
+    {
+        if(n==0)
```
Format the Code with spaces around the Operators

ErzaTitania-2001 committed:
Update StrongRecursion.java

Removed extra lines of code
Formatted space around operators

ErzaTitania-2001 reviewed (commented):
If any other problems present than indentation, do say @HarshCasper

ErzaTitania-2001 committed:
Update StrongRecursion.java

changed ; to :

iamrajiv reviewed (changes requested):
@ErzaTitania-2001 Rename the program to `check_strong_number.java`. Also, Format your code using code formatter.

ErzaTitania-2001 committed:
Update StrongRecursion.java

Changed class name
Formatted code using https://www.tutorialspoint.com/online_java_formatter.htm

iamrajiv reviewed (changes requested):
@ErzaTitania-2001 Rename the program `StrongRecursion.java` to `check_strong_number.java`.

ErzaTitania-2001 committed:
Rename StrongRecursion.java to check_strong_number.java

Changed class name

ErzaTitania-2001 committed:
Update README.md

iamrajiv committed:
Update check_strong_number.java

ErzaTitania-2001 commented:
@iamrajiv , should I close this pull request ? I want to pull another request on another program.

vatsalkesarwani12 reviewed (approved):
👍🏻

ErzaTitania-2001 committed:
Bucket Sort

Bucket Sort #1590

ErzaTitania-2001 committed:
Bucket Sort Using Hash

Bucket Sort #1590

ErzaTitania-2001 commented:
@HarshCasper  @iamrajiv , I made a commit for the issue Bucket Sort #1590 in the same pull request  by mistake. Any solutions to this problem?

iamrajiv reviewed (changes requested):
@ErzaTitania-2001 Remove the Bucket Sort program from this PR and make another PR for Bucket Sort. Also, whenever you make PR make separate branches for each PR rather than committing to the master branch.

ErzaTitania-2001 committed:
Delete Bucket_Sort.java

ErzaTitania-2001 commented:
@iamrajiv Done. Deleted Bucket_Sort. When is the PR gonna be merged?

iamrajiv committed:
Update README.md
